api/loaders/mysql_loader.py

```
@@ -54,33 +54,24 @@ class MySQLLoader(BaseLoader):
+        MySQL implementation using ORDER BY RAND() for random sampling.
         """
         query = f"""
-            SELECT COUNT(*) AS total_count,
-                COUNT(DISTINCT `{col_name}`) AS distinct_count
-            FROM `{table_name}`;
+            SELECT DISTINCT `{col_name}`
+            FROM `{table_name}`
+            WHERE `{col_name}` IS NOT NULL
+            ORDER BY RAND()
+            LIMIT %s;
         """
```

Ruff (0.14.10)

Possible SQL injection vector through string-based query construction

(S608)